public_html/wp-content/plugins/pattern-directory/includes/pattern-post-type.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace WordPressdotorg\Pattern_Directory\Pattern_Post_Type;
+
+const POST_TYPE = 'wporg-pattern';
+
+add_action( 'init', __NAMESPACE__ . '\register_post_type_data' );
+add_filter( 'rest_' . POST_TYPE . '_item_schema', __NAMESPACE__ . '\embed_extra_fields_in_search_endpoint' );
+add_action( 'enqueue_block_editor_assets', __NAMESPACE__ . '\enqueue_editor_assets' );
+
+
+/**
+ * Registers post types and associated taxonomies, meta data, etc.
+ */
+function register_post_type_data() {
+	register_post_type(
+		POST_TYPE,
+		array(
+			'label'        => 'Block Pattern',
+			'description'  => 'Stores publicly shared Block Patterns (predefined block layouts, ready to insert and tweak).',
+			'public'       => true,
+			'show_in_rest' => true,
+			'rewrite'      => array( 'slug' => 'pattern' ),
+
+			'supports' => array(
+				'title', 'editor', 'revisions', 'author', 'excerpt', 'thumbnail', 'custom-fields',
```

Comment:
If we're adding a separate meta field for the description, what's the use for the excerpt?

Reply:
🤔 that's a great point, we probably don't need it. I'll remove it.

Reply:
I went ahead and removed `thumbnail` and `revisions` too, since we're not using them (at least not yet).